src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -1641,33 +1641,40 @@ class spell_sha_nature_guardian : public SpellScriptLoader
             return ValidateSpellInfo({ SPELL_SHAMAN_NATURE_GUARDIAN });
         }
 
-        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        bool CheckProc(ProcEventInfo& eventInfo)
         {
-            PreventDefaultAction();
             DamageInfo* damageInfo = eventInfo.GetDamageInfo();
             if (!damageInfo || !damageInfo->GetDamage())
-                return;
+                return false;
+
+            int32 healthpct = GetSpellInfo()->Effects[EFFECT_1].CalcValue();
+            if (Unit* target = eventInfo.GetActionTarget())
+                if (target->HealthBelowPctDamaged(healthpct, damageInfo->GetDamage()))
+                    return true;
+
+            return false;
+        }
 
-            int32 healthpct = aurEff->GetSpellInfo()->Effects[EFFECT_1].CalcValue(); // %s2 - the 30% threshold for health
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            PreventDefaultAction();
 
             if (Unit* target = eventInfo.GetActionTarget())
             {
-                if (target->HealthBelowPctDamaged(healthpct, damageInfo->GetDamage()))
-                {
-                    CastSpellExtraArgs args(aurEff);
-                    args.AddSpellBP0(CalculatePct(target->GetMaxHealth(), aurEff->GetAmount()));
-                    target->CastSpell(target, SPELL_SHAMAN_NATURE_GUARDIAN, args);
-
-                    // Threat reduction is around 10% confirmed in retail and from wiki
-                    Unit* attacker = eventInfo.GetActor();
-                    if (attacker->IsAlive())
-                        attacker->GetThreatManager().ModifyThreatByPercent(target, -10);
-                }
+                CastSpellExtraArgs args(aurEff);
+                args.AddSpellBP0(CalculatePct(target->GetMaxHealth(), aurEff->GetAmount()));
+                target->CastSpell(target, SPELL_SHAMAN_NATURE_GUARDIAN, args);
+
+                // Threat reduction is around 10% confirmed in retail and from wiki
+                Unit* attacker = eventInfo.GetActor();
+                if (attacker->IsAlive())
+                    attacker->GetThreatManager().ModifyThreatByPercent(target, -10);
```

Comment:
There is a serverside spell for doing this (ID 39301), unsure of it's blizz or not.

Reply:
yes.
39301 should be casted there :)